src/platform/options/default_configuration.cpp

```diff
@@ -340,6 +341,60 @@ auto mo::DefaultConfiguration::options_for(SharedLibrary const& module) const ->
     return parsed_options;
 }
 
+namespace
+{
+std::string markdown_escape(std::string const& text)
+{
+    std::string escaped_text;
+
+    for (char const c : text)
+    {
+        switch(c)
+        {
+        case '&':
+            escaped_text += "&amp;";
+            break;
+        case '<':
+            escaped_text += "&lt;";
+            break;
+        case '>':
+            escaped_text += "&gt;";
+            break;
+        case '\"':
+            escaped_text += "&quot;";
+            break;
+        case '\'':
+            escaped_text += "&apos;";
+            break;
+        default:
+            escaped_text += c;
+            break;
+        }
+    }
+
+    return escaped_text;
+}
+
+std::string options_to_markdown(const boost::program_options::options_description& desc)
+{
+    std::ostringstream text;
+    for (auto& o: desc.options())
+    {
+        text << "### `" << o->format_name();
+        if (!o->format_parameter().empty())
+        {
+            text << " " << o->format_parameter();
+        }
+        text << "`" << std::endl;
+        text << std::endl;
+        text << markdown_escape(o->description()) << std::endl;
```

Comment:
It is needlessly inefficient to flush the output for every line:
```suggestion
        text << "`\n\n";
        text << markdown_escape(o->description()) << '\n';
```